src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -392,17 +401,47 @@ def relative_to_exec_root(path):
         fp.write(arg)
         fp.write('\n')
 
+    def maybe_execute_native_image():
+      if self.get_options().native_image_location is not None:
+        return self._compile_native_image(zinc_args)
+      else:
+        return self._compile_nonhermetic(jvm_options, zinc_args)
+
     return self.execution_strategy_enum.resolve_for_enum_variant({
       self.HERMETIC: lambda: self._compile_hermetic(
```

Comment:
Is it worth applying the same hack to the hermetic strategy in this patch? We'll need to be benchmarking with it soon.

Reply:
Based on the conversation this morning, I'll retract this comment, and suggest ensuring that we can provide native-images for both zinc and rsc in this patch.